spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/CassandraConnectionFactory.scala

```diff
@@ -77,9 +77,33 @@ object DefaultConnectionFactory extends CassandraConnectionFactory {
         } finally {
           IOUtils.closeQuietly(trustStoreFile)
         }
-
+        val kmf = if(conf.clientAuthEnabled){
+            conf.keyStorePath map {
+              case path ⇒
+                val keyStoreFile = new FileInputStream(path)
+                val kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm)
+                val keyStore = KeyStore.getInstance(conf.keyStoreType)
+                conf.keyStorePassword match {
```

Comment:
Looks like you repeating the same code twice, while only password should be mapped:
```scala
val keyStorePassword = conf.keyStorePassword.map(_.toCharArray)
keyStore.load...
kmf.init...
```